src/components/NewTodo.js

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import './NewTodo.css';
+import PropTypes from 'prop-types';
+
+class NewTodo extends React.Component {
+  state = {
+    id: this.props.id,
+    title: '',
+    completed: false,
+    user: {},
+    userId: '',
+    errorTitle: '',
+    errorUser: '',
+  }
+
+  handleChangeUser = (e) => { // choose user
+    this.setState({
+      user: this.props.users.find(user => user.id === +e.target.value),
+      userId: e.target.value,
+      errorUser: '',
+    });
+  }
+
+  handleChangeTitle = (e) => { // set title
+    this.setState({
+      title: e.target.value.replace(/[^a-z0-9\s]/gi, ''),
+      errorTitle: '',
+    });
+  }
+
+  handleChangeCheckbox = () => { // set checkbox
+    this.setState(prev => ({
+      completed: !prev.completed,
+    }));
+  }
+
+  resetForm = () => { // form reset
+    this.setState({
+      title: '',
+      completed: false,
+      userId: '',
+    });
+  }
+
+  handleSubmit = (e) => { // form submit
+    e.preventDefault();
+
+    if (this.state.title.length < 4) {
+      this.setState({
+        errorTitle:
+  <div className="todo__error">
+    Please enter the title
+  </div>,
+      });
+    }
+
+    if (this.state.userId === '') {
+      this.setState({
```

Comment:
there is not a good idea to use several setStates in one method, it's an async function and you can get an unexpected result;
write setState once and put all your validations inside it

Reply:
like this
```
this.setState((prevState) => {
  if (...) {...}
  if (...) {...}
})
```